runtime/metricsview/executor_rewrite_approx_comparisons.go

```diff
@@ -100,15 +113,17 @@ func (e *Executor) rewriteApproxComparisonNode(a *AST, n *SelectNode) bool {
 		n.FromSelect.Limit = a.Root.Limit
 		n.FromSelect.Offset = a.Root.Offset
 
-		// ---- CTE Optimization ---- //
-		// make FromSelect a CTE
-		a.convertToCTE(n.FromSelect)
+		if cteOptimization {
+			// make FromSelect a CTE
+			a.convertToCTE(n.FromSelect)
 
-		// now change the JoinComparisonSelect WHERE clause to use selected dim values from CTE
-		for _, dim := range n.FromSelect.DimFields {
-			dimName := a.dialect.EscapeIdentifier(dim.Name)
-			dimExpr := "(" + dim.Expr + ")" // wrap in parentheses to handle expressions
-			n.JoinComparisonSelect.Where = n.JoinComparisonSelect.Where.and(fmt.Sprintf("%[1]s IS NULL OR %[1]s IN (SELECT %[2]q.%[3]s FROM %[2]q)", dimExpr, n.FromSelect.Alias, dimName), nil)
+			// now change the JoinComparisonSelect WHERE clause to use selected dim values from CTE
+			// this necessitates that order of dims in from and comparison select are same
```

Comment:
Agian, this should be guaranteed, though I notice it's not very clear from the comments. Did you find a palce where this is not guaranteed?